dotCMS/src/main/java/com/dotmarketing/business/LayoutAPI.java

```diff
@@ -99,5 +99,14 @@
 	 * @param request
 	 * @return
 	 */
-  public Optional<Layout> resolveLayout(HttpServletRequest request);
+    public Optional<Layout> resolveLayout(HttpServletRequest request);
+
+
+	/**
+	 * Try to find the layout associated to a role
+	 * @param layout {@link Layout}
+	 * @param role   {@link Role}
+	 * @return Optional LayoutsRoles, present and LayoutsRoles identifier set if exists.
+	 */
+	public Optional<LayoutsRoles> findLayoutByRole(final Layout layout, final Role role);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'findLayoutByRole': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5164094)